jaq/src/main.rs

```diff
@@ -134,6 +134,15 @@ fn binds(cli: &Cli) -> Result<Vec<(String, Val)>, Error> {
         let s = s.to_owned();
         Ok((k.to_owned(), Val::Str(s.into())))
     });
+    let argjson = cli.argjson.iter().map(|(k, s)| {
+        let s = s.to_owned();
+        use hifijson::token::Lex;
+        let mut lexer = hifijson::SliceLexer::new(s.as_bytes());
+        let v = lexer
+            .exactly_one(Val::parse)
+            .map_err(|e| Error::Parse(format!("cannot parse {s} as JSON: {e}")));
+        Ok::<(std::string::String, Val), Error>((k.to_owned(), v?))
```

Comment:
My rust knowledge fails me here, compiler not happy about deriving type parameter E for Result without this. Also looking at the other code i guess maybe you would like to refactor this to share parsing of json somehow?